src/tests/e2e/complete-user-workflow.spec.tsx

```diff
@@ -0,0 +1,339 @@
+    // Create multiple tracks to test pagination
+    for (let i = 1; i <= 3; i++) {
+      // Create track
+      const createButton = page.getByTestId('create-track-button');
+      await createButton.click();
+
+      const titleInput = page.getByLabel('Назва треку');
+      const artistInput = page.getByLabel('Виконавець');
+      const genreSelect = page.getByTestId('genre-selector');
+
+      await titleInput.fill(`Pagination Test Track ${i}`);
+      await artistInput.fill(`Pagination Test Artist ${i}`);
+      
+      const genreOptions = await genreSelect.locator('option').count();
+      if (genreOptions > 1) {
+        await genreSelect.selectOption('Rock');
+      }
+
+      // Use getByTestId for submit button in modal
+      const submitButton = page.getByTestId('submit-button');
+      await submitButton.click();
+
+      // Wait for modal to close
+      const modalTitle = page.getByRole('heading', { name: 'Створити новий трек' });
+      await expect(modalTitle).not.toBeVisible({ timeout: 10000 });
+      
+      // Wait for toast to disappear
+      await page.waitForTimeout(1000);
+      await page.waitForTimeout(1000); // Wait for API call
+    }
+
+    // Test pagination if it exists
+    const pagination = page.getByTestId('pagination');
+    
+    if (await pagination.isVisible()) {
+      // Test next page
+      const nextButton = page.getByTestId('pagination-next');
+      if (await nextButton.isEnabled()) {
+        await nextButton.click();
+        await page.waitForTimeout(500); // Reduced timeout
+        
+        // Test previous page
+        const prevButton = page.getByTestId('pagination-prev');
+        if (await prevButton.isEnabled()) {
+          await prevButton.click();
+          await page.waitForTimeout(500); // Reduced timeout
+        }
+      }
+    }
+
+    // Clean up - delete created tracks
+    const trackItems = page.locator('[data-testid*="track-item"]').filter({ hasText: 'Pagination Test Track' });
+    const count = await trackItems.count();
+    
+    if (count > 0) {
+      // Select all pagination test tracks
+      for (let i = 0; i < count; i++) {
+        const trackItem = trackItems.nth(i);
+        const checkbox = trackItem.locator('input[type="checkbox"]');
+        if (await checkbox.isVisible()) {
+          await checkbox.check();
+          await page.waitForTimeout(100); // Small delay between checks
+        }
+      }
+
+      // Delete selected tracks
+      const bulkDeleteButton = page.getByTestId('bulk-delete-button');
+      if (await bulkDeleteButton.isVisible()) {
+        await expect(bulkDeleteButton).toBeVisible({ timeout: 5000 });
+        await bulkDeleteButton.click();
+
+        const confirmButton = page.getByTestId('confirm-button');
+        await confirmButton.click();
+
+        // Wait for toast to disappear
+        await page.waitForTimeout(1000);
+        await page.waitForTimeout(1000); // Wait for API call
+      }
```

Comment:
It would also be great to check that the tracks disappear after `bulkDelete`.

```suggestion
      const trackItemsDeleted = page.locator('[data-testid*="track-item"]').filter({ hasText: 'Pagination Test Track' });
      await expect(trackItemsDeleted.count === 0)
```